query/src/servers/http/clickhouse_handler.rs

```diff
@@ -181,38 +151,34 @@ pub async fn clickhouse_handler_post(
         .await
         .map_err(InternalServerError)?;
 
-    let mut sql = params.query;
+    let mut sql = params.query.unwrap_or_default();
 
-    // Insert into format sql
-    let (plan, input_stream) = if let Some((format, statements)) =
-        try_parse_insert_formatted(&sql, ctx.get_current_session().get_type())
-            .map_err(BadRequest)?
-    {
-        let plan = PlanParser::build_plan(statements, ctx.clone())
+    let (plan, format, input_stream) = if sql.is_empty() {
+        sql = body.into_string().await?;
+        let (plan, format) = PlanParser::parse_with_format(ctx.clone(), &sql)
             .await
-            .map_err(InternalServerError)?;
-        ctx.attach_query_str(&sql);
+            .map_err(BadRequest)?;
 
-        let input_stream = match format {
-            Format::NDJson => build_ndjson_stream(&plan, body).await.map_err(BadRequest)?,
-        };
-        (plan, Some(input_stream))
+        (plan, format, None)
     } else {
-        // Other sql
-        let body = body.into_string().await.map_err(BadRequest)?;
-        let sql = format!("{}\n{}", sql, body);
```

Comment:
@sundy-li  

no longer support `let sql = format!("{}\n{}", sql, body);` ?

Reply:
and assume insert query always let SQL part in query param and  DATA part in body?

Reply:
I think it's reasonable, if the sql is empty then the body is the whole sql.